berkeley-function-call-leaderboard/eval_checker/executable_python_function.py

```diff
@@ -865,7 +865,11 @@ def get_movie_rating(movie_name):
     url = "http://www.omdbapi.com/"
     params = {"t": movie_name, "apikey": api_key["OMDB-API-KEY"]}
     response = requests.get(url, params=params)
-    return response.json()["Rated"]
```

Comment:
We don't need to add try except here since there's a try except outside in executable category here https://github.com/ShishirPatil/gorilla/blob/main/berkeley-function-call-leaderboard/eval_checker/checker.py#L702-L716 that handles when "Rated" is not a key. 

And the API is expected to return a dict with Rated field like the following, if there isn't such field, it would be caught in API sanity check. 

```bash
>>> params = {"t": "Avatar", "apikey": api_key["OMDB-API-KEY"]}
>>> response = requests.get(url, params=params)
>>> response.json()
{'Title': 'Avatar', 'Year': '2009', 'Rated': 'PG-13', 'Released': '18 Dec 2009', 'Runtime': '162 min', 'Genre': 'Action, Adventure, Fantasy', 'Director': 'James Cameron', 'Writer': 'James Cameron', 'Actors': 'Sam Worthington, Zoe Saldana, Sigourney Weaver', 'Plot': 'A paraplegic Marine dispatched to the moon Pandora on a unique mission becomes torn between following his orders and protecting the world he feels is his home.', 'Language': 'English, Spanish', 'Country': 'United States, United Kingdom', 'Awards': 'Won 3 Oscars. 91 wins & 131 nominations total', 'Poster': 'https://m.media-amazon.com/images/M/MV5BZDA0OGQxNTItMDZkMC00N2UyLTg3MzMtYTJmNjg3Nzk5MzRiXkEyXkFqcGdeQXVyMjUzOTY1NTc@._V1_SX300.jpg', 'Ratings': [{'Source': 'Internet Movie Database', 'Value': '7.9/10'}, {'Source': 'Rotten Tomatoes', 'Value': '81%'}, {'Source': 'Metacritic', 'Value': '83/100'}], 'Metascore': '83', 'imdbRating': '7.9', 'imdbVotes': '1,397,142', 'imdbID': 'tt0499549', 'Type': 'movie', 'DVD': 'N/A', 'BoxOffice': '$785,221,649', 'Production': 'N/A', 'Website': 'N/A', 'Response': 'True'}
>>> response.json()["Rated"]
'PG-13'
```

Reply:
Thanks @CharlieJCJ . I have reverted the changes of handling exception at API level.  

The task of handling exceptions properly will be a blocker for evaluating this model. So, should merge these changes after that task is completed ?
 
One more thing to note is - There are already 10 instances in the `executable_python_function.py` file where exceptions are being handled at API level. Eg: https://github.com/ShishirPatil/gorilla/blob/main/berkeley-function-call-leaderboard/eval_checker/executable_python_function.py#L577

Reply:
I understand that correctly evaluating the model during your local evaluation have dependency on this new PR. This handling exception in the `get_executable_expected_output` logic will affect all models we want to review it thoroughly, related to #563. We can merge your model handler PR (which) first, and then merge the PR that addresses handling error PR (with API definition fixes) since this will affects all models' score, they are fairly separate sources of concerns. 

For context, this error didn't show up previously in our internal testing for a long time (over a couple months) and we have being actively evaluating recently as well using the omdb key, and didn't have this API exception.

> There are already 10 instances in the executable_python_function.py file where exceptions are being handled at API level.

For this, thanks for raising, we also noticed this and yes we should be consistent with other functions, and we will review all executable python functions to make sure there are no false negatives during evaluation.

Reply:
> I was getting KeyError when running the evaluation. Here are the logs:
> 
> ```
>   File "/gorilla/berkeley-function-call-leaderboard/eval_checker/eval_runner.py", line 495, in <module>
>     runner(model_names, test_categories, api_sanity_check)
>   File "/gorilla/berkeley-function-call-leaderboard/eval_checker/eval_runner.py", line 384, in runner
>     get_executable_expected_output(prompt_file)
>   File "/gorilla/berkeley-function-call-leaderboard/eval_checker/eval_runner_helper.py", line 908, in get_executable_expected_output
>     exec(
>   File "<string>", line 2, in <module>
>   File "/gorilla/berkeley-function-call-leaderboard/eval_checker/executable_python_function.py", line 880, in get_movie_director
>     return response.json()["Director"]
> KeyError: 'Director'
> ```

I faced same issue. resolved by updating the `omdbapi` key